### PR TITLE
Migrate from deprecated pedantic to flutter_lints

### LINKED ISSUE #471: [all] Migrate from deprecated pedantic to flutter_lints
## Proposal
Right now, all packages use pedantic. Pedantic however is discontinued. The Google team recommends using https://pub.dev/packages/flutter_lints for flutter packages. We should migrate all packages to this linter.

### PR DESCRIPTION
**Pull request by adityathakurxd**
## Description

Migrated all packages from deprecated `pedantic` to `flutter_lints`. 
Add a dev_dependency on package: `flutter_lints` to all project’s pubspec.yaml

Since the project already had a custom `analysis_options.yaml` file at its root, I added `include: package:flutter_lints/flutter.yaml` to it at the top

## Related Issues

closes #471

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by mhadaily** [comment]
Can you please merge with `main` thanks

**Comment by adityathakurxd** [comment]
I have resolved the conflicts to merge with main.

**Review by mhadaily — APPROVED** [review]
Build failed

**Comment by NatsuOnFire** [comment]
On some packages, you removed the sdk flutter from the integration_test dependency.
It is for this reason that the build fails

Need to specify the sdk flutter
```suggestion
  integration_test:
      sdk: flutter
```

File: packages/package_info_plus/package_info_plus/example/pubspec.yaml [hunk 1]
```diff
@@ -11,6 +11,7 @@ dependencies:
     sdk: flutter
   package_info_plus:
     path: ../
+  integration_test:
```
Comment (mhadaily) [inline packages/package_info_plus/package_info_plus/example/pubspec.yaml:14]: This is unwanted change it seems

**Comment by adityathakurxd** [comment]
Hey @NatsuOnFire, Thank you for pointing that out. I have added
```
  integration_test:
      sdk: flutter
```
to all files, it was removed from. Please @mhadaily if you could review the changes again.

Also, wanted to ask about this one file:
change

I haven't made any changes to it. Might be due to some new merge. Do I need to do something about it? If so, Could you help me with it?

**Comment by Elias8** [comment]
> I haven't made any changes to it. Might be due to some new merge. Do I need to do something about it? If so, Could you help me with it?

Please merge the main branch and update the PR.

Thanks!

**Comment by adityathakurxd** [comment]
Updated @Elias8. Let me know if it is okay now. Thanks!

File: packages/network_info_plus/network_info_plus/ios/Classes/getgateway.c [hunk 2]
```diff
@@ -13,8 +13,14 @@
...
-#include "route.h"
-#define TypeEN    "en0"
+#include "TargetConditionals.h"
+#if TARGET_IPHONE_SIMULATOR
+    #include <net/route.h>
+    #define TypeEN    "en1"
+#else
+    #include "route.h"
+    #define TypeEN    "en0"
+#endif
```
Comment (mhadaily) [inline packages/network_info_plus/network_info_plus/ios/Classes/getgateway.c:23]: This seems not to be in a correct PR. if you want to fix an issue, it would be nice if you can send another PR for this and mention the issue for it.

**Review by mhadaily — CHANGES_REQUESTED** [review]
It seems you have conflicts and the brand is not up to date again. I am sorry we are emerging so many PRs and that might happen.

**Comment by Elias8** [comment]
@mhadaily Did you see that some updated package versions are reverted to an older version? Did you pass intentionally?

**Comment by mhadaily** [comment]
@Elias8 what do you mean?

File: packages/android_alarm_manager_plus/pubspec.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name: android_alarm_manager_plus
 description: Flutter plugin for accessing the Android AlarmManager service, and
   running Dart code in the background when alarms fire.
-version: 1.3.0
+version: 1.2.0
```
Comment (Elias8) [inline packages/android_alarm_manager_plus/pubspec.yaml:4]: @mhadaily I think this is an unnecessary change. And there are other similar changes like this.

Comment (mhadaily) [reply packages/android_alarm_manager_plus/pubspec.yaml:4]: You are absolutely right. @adityathakurxd  I hope you will take care of it.

Comment (adityathakurxd) [reply packages/android_alarm_manager_plus/pubspec.yaml:4]: Yes, I am sorry about this. I'll take care of it and make it properly.

**Comment by adityathakurxd** [comment]
I'll close this and redo this taking care of the changes.

**Comment by mhadaily** [comment]
thanks, @adityathakurxd  looking forward to it
